tests/Endpoints/MultiSearchTest.php

```diff
@@ -90,4 +91,39 @@ public function testSupportedQueryParams(): void
         self::assertTrue($result['showRankingScore']);
         self::assertTrue($result['showRankingScoreDetails']);
     }
+
+    public function testMultiSearchWithDistinctAttribute(): void
+    {
+        $response = $this->client->multiSearch([
+            (new SearchQuery())->setIndexUid($this->booksIndex->getUid())
+                ->setQuery('princ')
+                ->setSort(['author:desc']),
+            (new SearchQuery())->setIndexUid($this->booksIndex->getUid())
+                ->setDistinct("genre")
+        ]);
+
+
+        self::assertCount(2, $response['results']);
+
+        self::assertArrayHasKey('indexUid', $response['results'][0]);
+        self::assertArrayHasKey('hits', $response['results'][0]);
+        self::assertArrayHasKey('query', $response['results'][0]);
+        self::assertArrayHasKey('limit', $response['results'][0]);
+        self::assertArrayHasKey('offset', $response['results'][0]);
+        self::assertArrayHasKey('estimatedTotalHits', $response['results'][0]);
+        self::assertCount(2, $response['results'][0]['hits']);
+
+        self::assertArrayHasKey('indexUid', $response['results'][1]);
+        self::assertArrayHasKey('hits', $response['results'][1]);
+        self::assertArrayHasKey('query', $response['results'][1]);
+        self::assertCount(4, $response['results'][1]['hits']);
```

Comment:
is the response dynamic, that you can't assert the whole structure at once?

Reply:
well, the main purpose of the test was that the hits in second search `(new SearchQuery())->setIndexUid($this->booksIndex->getUid())->setDistinct('genre')` don't have multiple genres since we only need distinct genres.

The first search is just for the purpose of multi-search and can be removed. 

As for whether the response is dynamic, it depends on how the distinct items are being chosen. 

@curquiza 
1. Is there some specific criteria for choosing the distinct records (First one in the list or random) ?
2. Is there any priority between the rules like do we apply limit after filtering or before it ?

I have tested on rc3 and it seems like it works by picking in FCFS manner. So, we can assert the response since it remains the same.

Also, @curquiza all documents not containing the distinct parameter are also returned. Is that the expected behavior ?

Reply:
@curquiza still waiting for response here :)

Reply:
oooops I missed this message!

> Is there some specific criteria for choosing the distinct records (First one in the list or random) ?
   Is there any priority between the rules like do we apply limit after filtering or before it ?

I really don't understand these questions, sorry, how they are related to distinct.

I would stay simple. Here we are in a SDK, we only test the SDK. You can test it as we test `distinctAttribute`, because `distinct` and `distinctAttribute` work exactly the same. One is during search time, the other one set in settings

So
- Add 2 documents with the same "genre"
- Add `genre` to `filterableAttribute`
- Search with `distinct: genre` (nothing else) -> ensure you get only 1 document

> Also, @curquiza all documents not containing the distinct parameter are also returned. Is that the expected behavior ?

Yes